### DISCUSSION TITLE
New version of the tier 3 template and new integration pattern

### DISCUSSION THREAD

**Discussion by brooke-hamilton** [Azure/missionlz#575] [Announcements]
We are modifying how we deploy add-ons to Mission Landing Zone, including the [tier 3 (new workload) template](https://github.com/Azure/missionlz/tree/main/src/bicep/examples/newWorkload). We are following the [shared variable file pattern](https://docs.microsoft.com/en-us/azure/azure-resource-manager/bicep/patterns-shared-variable-file) in which we generate a json file from the output of the Mission Landing Zone deployment, and that file is used by examples and add-ons to reference the state of the deployment.

Here's an example from the `newWorkload` [README.md](https://github.com/Azure/missionlz/tree/main/src/bicep/examples/newWorkload)

```BASH
# Generate a file named 'deploymentVariables.json' from a deployment of Mission Landing Zone named 'contoso'.
az deployment sub show -n contoso --query properties.outputs > ./deploymentVariables.json

# Deploy the `newWorkload` template.
az deployment sub create -n deployNewWorkload -f newWorkload.bicep -l eastus --parameters workloadName='newWorkload'
```

The `newWorkload.bicep` template has a reference to the `deploymentVariables.json` file generated above, and parameters have default values that come from the file. Here's a snippet from the `newWorkload` Bicep template:

```BICEP
param mlzDeploymentVariables object = json(loadTextContent('../deploymentVariables.json'))
...
param hubSubscriptionId string = mlzDeploymentVariables.hub.Value.subscriptionId
```

**Comment by LManning-Dev** [comment]
@brooke-hamilton Could you provide an example of how we would use a specific deploymentVariables file and send that into the mlzDeploymentVariables parameter? Hopefully in CLI and in PowerShell? I really appreciate it! I have tried Get-Content and ConvertFrom-Json but I am unable to get it to work. Thank you!
